From f7ce57047ceb18205ef9204897d14d2ce23d14fb Mon Sep 17 00:00:00 2001
From: "eldin.lee" <eldin.lee@quantatw.com>
Date: Wed, 9 Jul 2025 14:42:07 +0800
Subject: [PATCH 14/20] Add delay before leak event log to prevent AC OFF
 glitches

Insert a short delay before logging to prevent transient AC power loss from triggering incorrect events.
---
 src/leakdetector/LeakEvents.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/leakdetector/LeakEvents.cpp b/src/leakdetector/LeakEvents.cpp
index e3755a7..8cf8062 100644
--- a/src/leakdetector/LeakEvents.cpp
+++ b/src/leakdetector/LeakEvents.cpp
@@ -22,6 +22,8 @@ auto Events::generateLeakEvent(sdbusplus::message::object_path detectorPath,
 {
     auto eventName = std::make_tuple(detectorPath.str, level);
 
+    co_await sdbusplus::async::sleep_for(ctx, std::chrono::milliseconds(500));
+
     // NOLINTNEXTLINE(clang-analyzer-core.uninitialized.Branch)
     if (state == DetectorStateIntf::DetectorState::Normal)
     {
-- 
2.47.3

